lib/api/api.js

```diff
@@ -157,16 +162,35 @@ const api = {
                 // Convert array of post buffers into one string
                 request.post = Buffer.concat(post, postLength).toString();
 
-                if (apiMethod === 'objectCopy' ||
+                // IAM policy -Tag condition keys require information from CloudServer for evaluation
+                return tagConditionKeyAuth(authorizationResults, request, requestContexts, log,
+                (err, tagAuthResults) => {
+                    if (err) {
+                        log.trace('tag authentication error', { error: err });
+                        return callback(err);
+                    }
+                    if (tagAuthResults) {
+                        const checkedResults = checkAuthResults(tagAuthResults);
+                        if (checkedResults instanceof Error) {
+                            return callback(checkedResults);
+                        }
+                        returnTagCount = checkedResults;
+                    }
+                    if (apiMethod === 'objectPut' || apiMethod === 'objectPutPart') {
+                        return this[apiMethod](userInfo, request, streamingV4Params,
```

Comment:
I know you didn't make this change in particular, but using `this[apiMethod]` with varying parameters is quite confusing to me. Especially cause I am not even sure where the actual methods are ( I couldn't find them ). I am not sure a solution and I wouldn't block this PR cause of it, but definitely wanted to at least mention it.

Reply:
yeah, it's kind of confusing, but the api list is just below the big function in the same file, and each name is the imported method. they're all part of the same object, hence the `this`

Reply:
oh ok, I went up not down the file. The review was cut just before the methods.